### PR TITLE
RobinDict empty! not zeroing hashes bugfix

### PR DESCRIPTION
**Pull request by tozanski**
I believe without this change garbage keys from before `empty!` or garbage keys can be found in the dictionary which should be the case

### CODE REVIEW

**Comment by eulerkochy** [comment]
Hey! Can you explain your thought process a bit more elaborately ? 'Cuz I am unable to fathom the purpose of `fill`ing `hashes` with `0` or infact anything for that sake. Only after an successful `rh_insert`, the `hashes` table fills up with meaningful data. Letting it stay garbage was a choice I purposefully did while coding this up.

**Comment by tozanski** [comment]
The following code
```using DataStructures
dict = RobinDict{Int, Int}()

push!(dict, 2=>3)
haskey(dict, 1) |> display
haskey(dict, 2) |> display
empty!(dict)
haskey(dict, 1) |> display
haskey(dict, 2) |> display```
Produces unexpected output
```false
true
false
true```

**Comment by eulerkochy** [comment]
Cool ! Yes, it shouldn't be the case that `haskey` return `true` for any key!
Nice catch, btw ! Sharp eyes 🦅 !
Thanks for your contribution 😸

File: src/robin_dict.jl [hunk 1]
```diff
@@ -285,6 +285,7 @@ function empty!(h::RobinDict{K,V}) where {K, V}
     resize!(h.keys, sz)
     resize!(h.vals, sz)
     resize!(h.hashes, sz)
+    fill!(h.hashes, 0)
```
Comment (eulerkochy) [inline src/robin_dict.jl:288]: Add one/two test for this change.

**Comment by eulerkochy** [comment]
This was a very critical bug ! Thanks a ton @tozanski for correcting it !

**Comment by tozanski** [comment]
Thanks, I'm glad I could help

**Pull request merged**